lesson-demo/src/main/java/com/bobocode/pool/PooledDataSource.java

```diff
@@ -0,0 +1,33 @@
+package com.bobocode.pool;
+
+import lombok.SneakyThrows;
+import org.postgresql.ds.PGSimpleDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class PooledDataSource extends PGSimpleDataSource {
+    // todo: 1. store a queue of connections (this is a pool)
+    // todo: 2. initialize a datasource with 10 physical connection
+    // todo: 3. override method getConnection so it uses a pool
+    private BlockingQueue<Connection> pool;
+
+    @SneakyThrows
+    @Override
+    public Connection getConnection() {
```

Comment:
@NikolayProkopchuk it means that the fist user who needs to call the DB will have to wait for the pool initialisation. If you don't want to initialise max capacity at the beginning, you can introduce `min` and `max` pool size (E.g. `min=10`, `max = 20`).  Giving that you will initialise a pool with smaller amount of connections initially, and then will add new ones.  

So I would not recommend making it completely lazy.